### PR TITLE
Ansible Tower job_launch module

### PR DESCRIPTION
**Pull request by wwitzel3**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

Ansible Tower job_launch module.

##### ANSIBLE VERSION

```
ansible 2.3.0 (tower_module_job_launch 63442f69a4) last updated 2017/03/01 10:58:49 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY

Allows playbook authors to launch a Tower job.

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+- name: Launch a job
+  tower_job_launch:
+    job_template: "My Job Template"
+    register: job
+- name: Wait for job max 120s
+  tower_job_wait:
+    job_id: job.id
+    timeout: 120
+'''
+
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py:82]: 2017-03-01 16:33:44 ERROR:312:No RETURN documentation provided
2017-03-01 16:33:44 ERROR:107:Imports should be directly below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 152
2017-03-01 16:33:44 ERROR:107:Imports should be directly below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 153

See http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    job_template:
+      description:
+        - Name of the job_template to use.
+      required: True
+    job_explanation:
+      description:
+        - Job explanation field.
+      required: False
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py]: `required: False` lines can be removed, as it's the default.

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py (hunk 1 above)
Comment (wwitzel3) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py:82]: Updated, though not sure why github doesn't think I changed this part of the code.

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      default: null
+    limit:
+      description:
+        - Limit to use for the job_template.
+      required: False
+      default: null
+    tags:
+      description:
+        - Specific tags to use for from playbook.
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py]: Should this be of `type='list'`?

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            job = tower_cli.get_resource('job')
+
+            lookup_fields = ('job_template', 'inventory', 'credential')
+            for field in lookup_fields:
+                try:
+                    name = params.pop(field)
+                    result = tower_cli.get_resource(field).get(name=name)
+                    params[field] = result['id']
+                except exc.NotFound as excinfo:
+                    module.fail_json(msg='Unalbe to launch job, {0}({1}) was not found: {2}'.format(field, name, excinfo), changed=False)
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py]: Unalbe -> Unable

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py (hunk 3 above)
Comment (wwitzel3) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py]: We expect this to be a comma separated list, `foo, bar, baz` but internally we expect that list to be a string. So I guess we could set it to `type=list` and I can just `.join` it before I pass it in to our Tower library that makes the call.

**Review by gundalow — APPROVED** [review]

**Comment by ghjm** [comment]
shipit

**Comment by matburt** [comment]
shipit

**Comment by gundalow** [comment]
Shippable is green on `Sha: 43bbec3443` the Fedora24 job got stuck in git clone, though nothing to do with this PR.

**Review by gundalow — APPROVED** [review]

**Pull request merged**